dm/config/task.go

```diff
@@ -55,6 +56,8 @@ var (
 	// SyncerConfig
 	defaultWorkerCount = 16
 	defaultBatch       = 100
+
+	taskNameRegexp = regexp.MustCompile(`^[a-zA-Z0-9$_]+$`).MatchString
```

Comment:
better to remove the `MatchString` function signature or change the name of `taskNameRegexp`

Reply:
btw, will the restriction on task name affect the existing tasks?

Reply:
okay, removed `MatchString`, see [here](https://github.com/pingcap/dm/pull/290/files#diff-b263153ff75f30e0c0f3fd51160160faR334)

existing task won't failed, but if it stop start again, update, it will failed